packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -46,6 +62,64 @@ Map<MaterialListType, double> kListTileExtent = const <MaterialListType, double>
   MaterialListType.threeLine: 88.0,
 };
 
+/// An inherited widget that defines  color and style parameters for [ListTile]s
+/// in this widget's subtree.
+///
+/// Values specified here are used for [ListTile] properties that are not given
+/// an explicit non-null value.
+///
+/// The [Drawer] widget specifies a tile theme for its children which sets
+/// [style] to [ListTileStyle.drawer].
+class ListTileTheme extends InheritedWidget {
+  /// Creates an inherited widget that defines color and style parameters for [ListTile]s.
+  const ListTileTheme({
+    Key key,
+    this.dense: false,
+    this.style: ListTileStyle.normal,
+    this.selectedColor,
+    this.iconColor,
+    this.textColor,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  /// If true then [ListTile]s will have the vertically dense layout.
+  final bool dense;
+
+  /// If specified, [style] defines the font used for [ListTile] titles.
+  final ListTileStyle style;
+
+  /// If specified, the color used for icons and text when a [ListTile] is selected.
+  final Color selectedColor;
+
+  /// If specified, the icon color used for enabled [ListTile]s that are not selected.
+  final Color iconColor;
+
+  /// If specified, the text color used for enabled [ListTile]s that are not selected.
+  final Color textColor;
+
+  /// The closest instance of this class that encloses the given context.
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// ListTileTheme theme = ListTileTheme.of(context);
+  /// ```
+  static ListTileTheme of(BuildContext context) {
+    final ListTileTheme result = context.inheritFromWidgetOfExactType(ListTileTheme);
+    return result ?? new ListTileTheme();
+  }
+
+  @override
+  bool updateShouldNotify(ListTileTheme oldTheme) {
+    return dense != oldTheme.dense ||
```

Comment:
can you line this up so they all start at the same offset? I didn't notice dense at first because it looked like just four expressions. (Maybe use abarth style and put the || operators on the left under the return?)